lib/config/folderconfiguration.go

```diff
@@ -235,7 +235,7 @@ func (f *FolderConfiguration) SharedWith(device protocol.DeviceID) bool {
 	return false
 }
 
-func (f *FolderConfiguration) CheckAvailableSpace(req int64) error {
+func (f *FolderConfiguration) CheckAvailableSpace(req uint64) error {
```

Comment:
This breaks the check on L248-249, which is why we avoid unsigned integers to begin with. Change might still be desired, but requires careful auditing everywhere the values are used in any kind of arithmetic

Reply:
Thanks for catching. Corrected and checked that `fs.Usage` values aren't involved in any arithmetics anywhere else.